### DISCUSSION TITLE
What module constitutes the actual linting?

### DISCUSSION THREAD

**Discussion by deostroll** [typescript-eslint/typescript-eslint#7485] [Technical Discussions]
### Body

Body of the technical discussion...
Just trying to generally understand how linting works. Hence would like to know modules involved. I have superficial knowledge; which is equivalent to no-knowledge at all sometimes. 😖

I understand something has to create AST. But from AST to linted source is the aspect I want to understand more. Hoping someone can provide a nice starting point to help me understand this.

### Additional Info

Any additional info...

### Before you submit your discussion, please confirm the following. If any of these required steps are not taken, we may not be able to review your RFC. Help us to help you!

- [X] I have [read the discussions guidelines](https://typescript-eslint.io/contributing/discussions) and per those guidelines, this fits a discussion category, _not_ a help request or standard issue.
- [X] I have [searched for related discussions](https://github.com/typescript-eslint/typescript-eslint/discussions) and [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

**Comment by bradzacher** [comment]
As per our contributing guidelines (which you ticked) - we don't use discussions for q&a

Please read through for alternatives.
https://typescript-eslint.io/contributing/discussions
